contact_map/contact_map.py

```diff
@@ -864,14 +889,95 @@ def from_contacts(self, *args, **kwargs):  #pylint: disable=W0221
     def atom_contacts(self):
         n_x = self.topology.n_atoms
         n_y = self.topology.n_atoms
-        diff = collections.Counter(self.positive.atom_contacts.counter)
-        diff.subtract(self.negative.atom_contacts.counter)
+        filtered_pos = self.positive.atom_contacts.filter(
+            self._all_atoms_intersect)
+        filtered_neg = self.negative.atom_contacts.filter(
+            self._all_atoms_intersect)
+
+        diff = collections.Counter(filtered_pos.counter)
+        diff.subtract(filtered_neg.counter)
         return ContactCount(diff, self.topology.atom, n_x, n_y)
 
     @property
     def residue_contacts(self):
         n_x = self.topology.n_residues
         n_y = self.topology.n_residues
-        diff = collections.Counter(self.positive.residue_contacts.counter)
-        diff.subtract(self.negative.residue_contacts.counter)
+        filtered_pos = self.positive.residue_contacts.filter(
+            self._all_residues_intersect)
+        filtered_neg = self.negative.residue_contacts.filter(
+            self._all_residues_intersect)
+        diff = collections.Counter(filtered_pos.counter)
+        diff.subtract(filtered_neg.counter)
         return ContactCount(diff, self.topology.residue, n_x, n_y)
```

Comment:
Huh... apparently approving in CodeClimate is for the PR (forever), not for the commits. I was surprised CodeClimate didn't complain about repeated code -- but actually, it does.

Anyway, doesn't necessarily need to be fixed in this PR -- up to you. We could also handle it in a later reduce-technical-debt PR.

Reply:
I am in favor of reducing technical dept as soon as possible (these things have the tendency to keep staying at the bottom of my priority list otherwise :laughing: ) Should be better now.

Reply:
CodeClimate still does not like it even when they are completely factored out...

Reply:
I've marked the CC complaint as "invalid." The "code mass" that defines a cutoff for similarity is [basically just the number of nodes in the AST](https://docs.codeclimate.com/docs/duplication#section-understand-the-engine). If we have too many problems with this, we can edit our `.codeclimate.yml` -- actually, that kind of looks like it isn't correct right now anyway, since it has a `Similar code` line instead of `similar-code` as in the docs.

I may post an issue on the repo for the [duplication plugin](https://github.com/codeclimate/codeclimate-duplication). Everything behaves as advertised, but only if you read the fine print. There's some unintuitive behavior here. For example, I suspect that simply not using keywords might make this go away (keywords add more nodes to the AST than regular args). Also, dropping kwargs from only one of the functions would probably make it so CC doesn't notice the similarity, which seems like a lack of functionality.

That said, there is a way around this. I think this is part of why I originally made the `ContactsDict` object. With that, you can do `self.positive.contacts['res']` to get the residue contacts. Since the generalized function has access to `self`, you can move all the parameters into a dictionary in the generalized function, and then just pass `'atom'` or `'res'` to that function and select the right thing from the dict.

Anyway, will merge as is and we can do that later if we desire. I do not think the current similarity adds a significant maintenance burden.